src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -492,6 +506,23 @@ private static String getCourseAPlusUrl(@NotNull JSONObject jsonObject, @NotNull
     return replInitialCommands;
   }
 
+  @NotNull
+  private static Version getCourseVersion(@NotNull JSONObject jsonObject,
+                                                @NotNull String source)
+      throws MalformedCourseConfigurationException {
+    String versionJson = jsonObject.optString("version", null);
+    if (versionJson == null) {
+      return BuildInfo.INSTANCE.courseVersion;
```

Comment:
Does that effectively mean that if version is missing from course config file, it is expected to be fully compliant by default.

Reply:
Yes, if there's no information about version in the course configuration file, then no version check should be performed. This is achieved by making the course version always the same as the user's version (= always fully compliant, as you said)